api/src/wfl/api/workloads.clj

```diff
@@ -58,14 +60,16 @@
   [_ body]
   (throw
     (ex-info "Failed to create workload - no such pipeline"
-      (select-keys body [:pipeline]))))
+      {:cause (:pipeline body)
+       :type ::invalid-workload})))
```

Comment:
Should we throw the entire `body` with the exception?

Reply:
That will give the client
```json
{
    "data": {
        "cause": {
            "creator": "wfl-non-prod@broad-gotc-dev.iam.gserviceaccount.com",
            "cromwell": "https://cromwell-gotc-auth.gotc-dev.broadinstitute.org",
            "input": "fake-input",
            "items": [
                {
                    "inputs": {
                        "dst": "gs://broad-gotc-dev-zero-test/wfl-test-b17d834c-7c27-4e7a-a6df-3e2a4ef4def7/output.txt",
                        "src": "gs://broad-gotc-dev-zero-test/wfl-test-b17d834c-7c27-4e7a-a6df-3e2a4ef4def7/input.txt"
                    }
                }
            ],
            "output": "fake-output",
            "pipeline": "bogus",
            "project": "(Test) Exhaustive WFL Code Bashing Test"
        },
        "type": "wfl.api.workloads/invalid-workload"
    },
    "exception": "clojure.lang.ExceptionInfo",
    "message": "Invalid Workload",
    "uri": "/api/v1/create"
}
```
rather than
```json
{
    "data": {
        "cause": "bogus",
        "type": "wfl.api.workloads/invalid-workload"
    },
    "exception": "clojure.lang.ExceptionInfo",
    "message": "Invalid Workload",
    "uri": "/api/v1/create"
}
```
more information but less focus on the error field. What do you think?

Reply:
Thank you for this.  I think you understand the trade-off.  I'll leave it up to you.
FWIW: I like the extra information and it looks OK to return to the client.

Reply:
Thanks! I will change it so it returns the extra info, and I also think it makes sense to update the exception type to `Invalid Pipeline` to be more specific.

Reply:
Update per our discussion.